File: src/js/directable/directables/dotmatrix/grid/GridRenderer.cs
using UnityEngine;

namespace UnityTest.UserInterface.DotMatrix.Grid
{
    public class GridRenderer
    {
        private GameObject m_go;
        private GridRendererComponent m_gridRendererComponent;

        public GridRenderer(GameObject container)
        {
            // Create GameObject
            m_go = new GameObject("GridRenderer");
            m_go.transform.SetParent(container.transform);

            // Create Component
            m_gridRendererComponent = m_go.AddComponent<GridRendererComponent>();
        }

        public void SetResolution(Vector2Int resolution)
        {
            m_gridRendererComponent.SetResolution(resolution);
        }

        // public void Update()
        // {
        // }
    }
}

File: src/js/directable/dotmatrix/grid/GridRenderer.js


namespace UnityTest.UserInterface.DotMatrix.Grid
{
    export default class GridRenderer
    {
        private GameObject #go;
        private GridRendererComponent #gridRendererComponent;

        GridRenderer(GameObject container)
        {
            // Create GameObject
            #go = new GameObject("GridRenderer");
            #go.transform.SetParent(container.transform);

            // Create Component
            #gridRendererComponent = #go.AddComponent<GridRendererComponent>();
        }

        SetResolution(Vector2Int resolution)
        {
            #gridRendererComponent.SetResolution(resolution);
        }

        // Update()
        // {
        // }
    }
}
